ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/rules/HiveRelFieldTrimmer.java

```diff
@@ -203,6 +202,25 @@ protected RexNode handle(RexFieldAccess fieldAccess) {
     return dispatchTrimFields(input, fieldsUsedBuilder.build(), extraFields);
   }
 
+  @Override
+  protected void setColumnAccessInfoForViews(RelNode rel, ImmutableBitSet fieldsUsed) {
+    final ColumnAccessInfo columnAccessInfo = COLUMN_ACCESS_INFO.get();
+    final Map<RelNode, Table> relNodeToTableAndProjects = VIEW_RELNODE_TO_TABLE.get();
+
+    // HiveTableScans are handled separately in HiveTableScan's trimFields method.
+    if (!(rel instanceof HiveTableScan) &&
+        columnAccessInfo != null &&
+        relNodeToTableAndProjects != null &&
+        relNodeToTableAndProjects.containsKey(rel)) {
+      Table table = relNodeToTableAndProjects.get(rel);
+      List<FieldSchema> tableAllCols = table.getAllCols();
+      
+      for (int i = fieldsUsed.nextSetBit(0); i >= 0; i = fieldsUsed.nextSetBit(i + 1)) {
```

Comment:
It's a bit hard to understand what the for-loop does without going and reading the Javadoc of `ImmutableBitSet#nextSetBit`. For readability, it would be probably much easier to understand the following:
```java
for (Integer i: fieldsUsed) {
```

Reply:
Updated the code here. Checked that the iterator is also using `ImmutableBitSet#nextSetBit`, but this is definitely more easy to understand.